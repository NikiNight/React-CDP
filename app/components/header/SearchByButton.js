import React from 'react';

const SearchByButton = (props) => {
    return(
        <button className={`button ${props.isActive ? 'button_active' : ''}`} 
                value={props.value} 
                onClick={() => props.toggleButtonsState(props.queryValue)}
        >
            {props.value}
        </button>
    )
}

export default SearchByButton;