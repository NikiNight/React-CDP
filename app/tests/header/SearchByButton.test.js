import React from 'react';
import renderer from 'react-test-renderer';
import SearchByButton from './../../components/header/SearchByButton';
import {mount} from 'enzyme';

describe('Test SearchByButton component', () => {

    test('matches snapshot', () => {
        const tree = renderer.create(<SearchByButton isActive={true} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should be call toggleButtonsState', () => {
        const toggleButtonsState = jest.fn();
        const component = mount(<SearchByButton toggleButtonsState={toggleButtonsState}/>);
        component.find('.button').simulate('click');
        expect(toggleButtonsState).toHaveBeenCalled();
    });

})