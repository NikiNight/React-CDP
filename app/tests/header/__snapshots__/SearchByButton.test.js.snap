// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchByButton component matches snapshot 1`] = `
<button
  className="button button_active"
  onClick={[Function]}
/>
`;
